plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaTable.java

```diff
@@ -0,0 +1,180 @@
+    {
+        this.columns = new ArrayList<>(columns);
+        this.constraints = ImmutableList.copyOf(constraints);
+    }
+
+    public void addColumn(String name, Object type, boolean nullable, Optional<String> comment, Map<String, Object> metadata)
+    {
+        checkArgument(columns.stream().noneMatch(column -> column.name.equalsIgnoreCase(name)), "Column already exists: %s", name);
+        columns.add(new Column(name, type, nullable, comment, metadata, null));
+    }
+
+    public void removeColumn(String name)
+    {
+        Column column = findColumn(name);
+        boolean removed = columns.remove(column);
+        checkState(removed, "Failed to remove '%s' from %s", name, columns);
+    }
+
+    public void renameColumn(String source, String target)
+    {
+        checkArgument(columns.stream().noneMatch(column -> column.name.equalsIgnoreCase(target)), "Column already exists: %s", target);
+
+        Column column = findColumn(source);
+        int index = columns.indexOf(column);
+        verify(index >= 0, "Unexpected column index");
+
+        Column newColumn = new Column(target, column.type, column.nullable, column.comment, column.metadata, column.generation);
+        columns.set(index, newColumn);
+    }
+
+    public List<String> columnNames()
+    {
+        return columns.stream()
+                .map(column -> column.name)
+                .collect(toImmutableList());
+    }
+
+    @NotNull
+    public Object getColumnType(String name)
+    {
+        return findColumn(name).type;
+    }
+
+    public boolean isNullable(String name)
+    {
+        return findColumn(name).nullable;
+    }
+
+    @Nullable
+    public String getColumnComment(String name)
+    {
+        return findColumn(name).comment.orElse(null);
+    }
+
+    public void setColumnComment(String name, Optional<String> comment)
+    {
+        requireNonNull(comment, "comment is null");
+        Column oldColumn = findColumn(name);
+        Column newColumn = new Column(oldColumn.name, oldColumn.type, oldColumn.nullable, comment, oldColumn.metadata, oldColumn.generation);
+        columns.set(columns.indexOf(oldColumn), newColumn);
+    }
+
+    @Nullable
+    public Map<String, Object> getColumnMetadata(String name)
+    {
+        return findColumn(name).metadata;
+    }
+
+    @Nullable
+    public String getColumnGeneration(String name)
+    {
+        return findColumn(name).generation;
+    }
+
+    public List<String> constraints()
+    {
+        return constraints;
+    }
+
+    private Column findColumn(String name)
+    {
+        requireNonNull(name, "name is null");
+        return columns.stream().filter(column -> column.name.equals(name)).collect(onlyElement());
+    }
+
+    @SuppressWarnings("UnusedVariable")
+    private record Column(String name, Object type, boolean nullable, Optional<String> comment, @Nullable Map<String, Object> metadata, @Nullable String generation)
```

Comment:
I don't follow why `@Nullable String generation`.
What speaks against `Optional<String> generation` (same as for `comment` parameter)?

Actually I see that the comment getter is also `@Nullable` . Can we make it to return `Optional<String>` instead?